src/components/Calendar/Section.js

```diff
@@ -36,28 +38,69 @@ export const styles = {
     justifyContent: 'space-between',
     flexWrap: 'wrap',
   },
+  modal: {
+    position: 'absolute',
+    width: '80%',
+    outline: 'none',
+    top: '12.5%',
+    left: '12.5%',
+    padding: '10px',
+  },
 };
 
-function Section({ classes, section }) {
-  return (
-    <Paper className={classes.paper}>
-      <div className={classes.container}>
-        <div className={classes.header}>
-          <Typography variant="caption" className={classes.text}>
-            {getFormattedClassEvent(section.event)}
-          </Typography>
+class Section extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+    };
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  toggleModal() {
+    const { showModal } = this.state;
+    this.setState({
+      showModal: !showModal,
+    });
+  }
+
+  render() {
+    const { classes, section } = this.props;
+    const { showModal } = this.state;
+    return (
+      <div>
+        <Paper className={classes.paper} onClick={this.toggleModal}>
+          <div className={classes.container}>
+            <div className={classes.header}>
+              <Typography variant="caption" className={classes.text}>
+                {getFormattedClassEvent(section.event)}
+              </Typography>
 
-          <Typography variant="caption" className={classes.text}>
-            {`${section.subjectId} ${section.courseId}`}
-          </Typography>
-        </div>
+              <Typography variant="caption" className={classes.text}>
+                {`${section.subjectId} ${section.courseId}`}
+              </Typography>
+            </div>
 
-        <Typography variant="subtitle2" className={`${classes.text} ${classes.name}`} noWrap>
-          {section.name}
-        </Typography>
+            <Typography variant="subtitle2" className={`${classes.text} ${classes.name}`} noWrap>
+              {section.name}
+            </Typography>
+          </div>
+        </Paper>
+        <Modal
```

Comment:
Like we talked about in the meeting, this modal could be its own component that gets imported into this component.

Reply:
@albirdy98 maybe you can give this a shot?

Reply:
Also, I don't think we need to be using the Modal component - the Modal component is a lower level component that is wrapped by the [Dialog](https://material-ui.com/demos/dialogs/) component of Material-UI, which I think we should use instead.